src/audio/sink_source_utils.c

```diff
@@ -72,4 +72,69 @@ int source_to_sink_copy(struct sof_source *source,
 	sink_commit_buffer(sink, INT_MAX);
 	return 0;
 }
+
+int sink_fill_with_silence(struct sof_sink *sink, size_t size)
+{
+	uint8_t *dst_ptr;
+	uint8_t *dst_begin;
+	uint8_t *dst_end;
+	size_t dst_size;
+	int ret;
+
+	if (!size)
+		return 0;
+	if (size > sink_get_free_size(sink))
+		return -ENOSPC;
+
```

Comment:
I would say `EINVAL` here.

Reply:
it is the same as in source_to_sink_copy - NO_SPACE error if somebody wants to copy/fill more than available space in destination